### PR TITLE
add qname feature to samtools/view

### PR DESCRIPTION
**Pull request by priyanka-surana**
`--qname` or `-N` flag with a read names list allows `samtools/view` to filter alignments

I could not figure out a test for the datasets available.

## PR checklist

Closes #XXX 

- [x ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, include test data in your PR.
- [ ] Remove all TODO statements.
- [ ] Emit the `versions.yml` file.
- [x ] Follow the naming conventions.
- [ x] Follow the parameters requirements.
- [ x] Follow the input/output options guidelines.
- [ ] Add a resource `label`
- [ ] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - [ ] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [x ] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [ ] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`

### CODE REVIEW

**Comment by nvnieuwk** [comment]
You can make an artificial file containing read names using `Channel.of(...readnames...).collectFile(name:readnames.list)` This way you can test this feature, assuming the file is just a file with one `readname` per line

File: modules/samtools/view/meta.yml [hunk 1]
```diff
@@ -33,6 +33,10 @@ input:
       type: optional file
       description: Reference file the CRAM was created with
       pattern: "*.{fasta,fa}"
+  - qname:
+      type: optional file
```
Comment (jfy133) [inline modules/samtools/view/meta.yml:37]:
```suggestion
      type: file
```

File: modules/samtools/view/meta.yml [hunk 2]
```diff
@@ -33,6 +33,10 @@ input:
...
+      description: File with read names to output only select alignments
```
Comment (jfy133) [inline modules/samtools/view/meta.yml:38]:
```suggestion
      description: Optional file with read names to output only select alignments
```

**Review by jfy133 — APPROVED** [review]
LGTM if tests passing

**Review by nvnieuwk — APPROVED** [review]

**Comment by priyanka-surana** [comment]
Added tests for the `qname` filtering.
FYI: `--write-index` does not work with `--output-fmt sam`

**Pull request merged**